lib/verdict/storage/cookie_storage.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Verdict
+  module Storage
+    # CookieStorage, unlike other Verdict storage classes, is distributed and stored on the client.
+    # Because cookies are opaque to users, we obsfucate information about the test such as the
+    # human readable names for the experiment or assignment group. This means this class assumes
+    # that `name` will be an obsfucated value, or one that we comfortable being "public".
+    class CookieStorage < BaseStorage
+      DEFAULT_COOKIE_LIFESPAN_SECONDS = 15778476 # 6 months
+
+      attr_accessor :cookies
+      attr_reader :cookie_lifespan
+
+      def initialize(cookies: {}, cookie_lifespan: DEFAULT_COOKIE_LIFESPAN_SECONDS)
+        @cookies = cookies
+        @cookie_lifespan = cookie_lifespan
+      end
+
+      def store_assignment(assignment)
+        hash = { group: assignment.group.name, created_at: assignment.created_at.strftime('%FT%TZ') }
+        set("verdict_#{assignment.experiment.name}", nil, JSON.dump(hash))
```

Comment:
Do we care about leaking our experiment names this way? A simple way to prevent this would be to use a digest.

Reply:
I highlighted this [above](https://github.com/Shopify/verdict/pull/51/files/37e601b2e2c0947704dbae40a4759a8bc7ff7d5e#diff-1c759bc412c1b63fd4a5f045f2066830R5-R8), but using a digest is clever too, I didn't think of that. I'll explore that path as having a reliance on `name` is awkward I agree.

Reply:
Updated to use digests, thanks for the tip!